pkg/controller/job/job_controller.go

```diff
@@ -1322,7 +1351,12 @@ func (jm *Controller) recordJobFinished(job *batch.Job, finishedCond *batch.JobC
 			jm.recorder.Event(job, v1.EventTypeWarning, "TooManySucceededPods", "Too many succeeded pods running after completion count reached")
 		}
 		jm.recorder.Event(job, v1.EventTypeNormal, "Completed", "Job completed")
+		// The succeeded metric with an empty reason is for the backward compatibility.
 		metrics.JobFinishedNum.WithLabelValues(completionMode, "succeeded", "").Inc()
+		if feature.DefaultFeatureGate.Enabled(features.JobSuccessPolicy) {
```

Comment:
As discussed in the other thread I think it is safe to drop it for now. Metrics are requirement for Beta only. Then, we could have a proper discussion in the KEP update if we want to modify `JobFinishedNum`.

Reply:
I see. I will drop all metrics implementations.

Reply:
Done.